addons/hitreactions/XEH_postInit.sqf

```diff
@@ -70,6 +70,14 @@
             (_this select 1) call CBA_fnc_removePerFrameHandler;
 
             _unit action ["TakeWeapon", _thrownWeapon, _weapon];
+
+            // "TakeWeapon" action bugs out unit where they don't switch weapons
+            // Resetting loadout fixes bugged state
+            [{
+                (_this select 0) hasWeapon (_this select 1)
+            }, {
+            	(_this select 0) setUnitLoadout getUnitLoadout (_this select 0);
```

Comment:
`setUnitLoadout` will break all things using unique magazines (intel items)

Reply:
> Do we have any other way of doing this?

I tried with `CBA_fnc_addWeaponWithoutItems` and adding items individually with `addWeaponItem`, but that didn't work, the AI would reload when receiving the new weapon.

I can try to use `addWeapon` and replacing the magazine that's inserted into the gun, but maybe even then the AI will reload, but I haven't checked yet.

> `setUnitLoadout` will break all things using unique magazines (intel items)

I wasn't aware of this because I added it in the arsenal. It's executed when you change a container (uniform, vest or backpack).

Reply:
I'm working on a method that doesn't use `setUnitLoadout`, but it's currently 100 extra lines of code.

Reply:
> `setUnitLoadout` will break all things using unique magazines (intel items)

Do we even care about that for AI?

Reply:
Fair - I have come up with a far better solution now.

Reply:
I retested it without the fix, just to make sure it was necessary.
I retested it several times, with several game restarts and it's now not necessary. I'm certain that yesterday's testing showed it was necessary, so I'm at a loss for words.